src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -515,7 +515,7 @@ void ParticipantConfiguration::finishParticipantConfiguration(
 
   for (impl::DataContext &dataContext : participant->readDataContexts()) {
     int toMeshID = dataContext.getMeshID();
-    PRECICE_CHECK(participant->isMeshProvided(toMeshID),
+    PRECICE_CHECK(participant->isMeshProvided(toMeshID) || participant->isDirectAccessAllowed(toMeshID),
```

Comment:
In which case is a mesh provided but cannot be accessed directly by the same participant?

Reply:
In case it was not configured. We require an explicit configuration of this feature at the moment. Otherwise people most probably do things the wrong way.

Reply:
Thanks, got it now. This is required by #1033.